app/assets/javascripts/components/manifest/manifest_view.jsx

```diff
@@ -1,127 +1,190 @@
+      if(consignment && consignment[element.name]){
+        element_result = consignment[element.name];
       }
-    ].filter(_=>_)
 
-    return (
+      let props = {
+        ...element,
+        result: element_result,
+        view_mode: this.state.view_mode
+      };
+
+      return(
+        <li key={index}>
+
+          <ManifestElement {...props} />
+        </li>
+      );
+    });
+
+    return <ol>{manifest_elements}</ol>;
+  }
+
+  renderPlots(){
+    if(this.state.view_mode != 'output') return null;
+
+    let {manifest} = this.props;
+    let plots = this.props.plots.map((plot, index)=>{
+      return(
+        <li key={plot.id}>
+
+          <a href={plotIndexUrl({manifest_id: manifest.id, id: plot.id})}>
+
+            {plot.name}
+          </a>
+        </li>
+      );
+    });
+
+    return(
+      <div>
+
+        <div>{'Plots: '}</div>
+        <ul>{plots}</ul>
+      </div>
+    );
+  }
+
+  render(){
+    return(
       <div className='manifest'>
+
         <div className='manifest-elements'>
-          <ButtonBar className='actions' buttons={ buttons }/>
-          <ManifestPreview {...manifest} />
-          <ToggleSwitch 
-            id="view_mode_switch"
-            caption="Show"
-            onChange={ (view_mode) => {
-              this.setState({ view_mode })
-              if (view_mode == 'output' && !consignment) {
-                if (!consignment) requestConsignments([manifestToReqPayload(manifest)])
-              }
-            } }
-            selected={this.state.view_mode}
-            values={[ 'script', 'output' ]}
-          />
-          <ol>
-            {manifestElements}
-          </ol>
-          {this.state.view_mode === 'output' &&
-            <div>
-              <div>{'Plots: '}</div>
-              <ul>
-                {this.props.plots.map(plot => (
-                  <li key={plot.id}>
-                    <a href={plotIndexUrl({ manifest_id: manifest.id, id: plot.id })}>
-                      {plot.name}
-                    </a>
-                  </li>
-                ))}
-              </ul>
-            </div>
-          }
+
+          {this.renderButtonBar()}
+          <ManifestPreview {...this.props.manifest} />
+          {this.renderToggleSwitch()}
+          {this.renderManifestElements()}
+          {this.renderPlots()}
         </div>
       </div>
-    )
+    );
   }
 }
 
-export default connect(
-  (state, props) => ({
-    consignment: selectConsignment(state, props.manifest.name),
-    plots: getPlotsByIds(state, props.manifest.plotIds || [])
-  }),
-  {
-    deleteManifest,
-    copyManifest,
-    toggleEdit,
-    requestConsignments,
-    selectPlot,
-    plotEdit
-  }
-)(ManifestView)
+const mapStateToProps = (state = {}, own_props)=>{
+  return {
+    consignment: selectConsignment(state, own_props.manifest.name),
+    plots: getPlotsByIds(state, own_props.manifest.plotIds || [])
+  };
+};
+
+const mapDispatchToProps = (dispatch, own_props)=>{
```

Comment:
Why favor this expanded form rather than the {} shorthand?

Reply:
Catching the dispatch at the component rather than at a returned function some where down in the stack (when that return function is finally run) is much easier to debug.

Reply:
My simple rule for design, UX, and usability in general is thus: If some required task is difficult to do then it requires a better design. Sometimes that requires elaboration to expose detail, sometimes that requires simplification to remove complexity. But the trigger is the difficulty in accomplishing a required task.

Reply:
I prefer the {} pattern because I think it encourages keeping the action dispatch simple, using the given action methods, rather than wrapping them in elaborate bespoke functions specific to this component in the connector (e.g., use 'requestDocuments' rather than requestSpecificDocument: (doc) => requestDocuments(...)). With a mapDispatchToProps = (dispatch,props) => {} i think the temptation to do the latter is greater.

I also don't think debugging requirements should dictate the form of the code. In this case it would be simple to trap the method when it is called in the component (e.g. this.requestDocuments() { this.props.requestDocuments }), and I'd rather write two lines of code to facilitate this if and when I have to debug this behavior rather than have it blown out into a function all of the time - this is a lot of boilerplate to support the theoretical requirement of debugging.

Anyway, just my $.02, much of this conversation is probably moot as we'll have to change all of these patterns anyway in order to decouple the component.

Reply:
Well, I'm not insisting that we 'must' do it one way or the other. But in the situations where I need to do it, I do it.